diff --git a/readconf.c b/readconf.c
index 8da57731d5..e2374a857e 100644
--- a/readconf.c
+++ b/readconf.c
@@ -2558,12 +2558,12 @@ parse_jump(const char *s, Options *o, int active)
 
 		if (first) {
 			/* First argument and configuration is active */
-			if (parse_ssh_uri(cp, &user, &host, &port) == -1 ||
+			if (parse_ssh_uri(cp, &user, &host, &port) == -1 &&
 			    parse_user_host_port(cp, &user, &host, &port) != 0)
 				goto out;
 		} else {
 			/* Subsequent argument or inactive configuration */
-			if (parse_ssh_uri(cp, NULL, NULL, NULL) == -1 ||
+			if (parse_ssh_uri(cp, NULL, NULL, NULL) == -1 &&
 			    parse_user_host_port(cp, NULL, NULL, NULL) != 0)
 				goto out;
 		}
